### ISSUE TITLE
Sync BN may cause Nan loss on tensorflow>=1.13

### ISSUE THREAD

**Issue by tabrisweapon** [tensorpack/tensorpack#1302]
First of all, Thanks for your wonderful work!

Based on your implementation, I write a sync_bn.batch_norm module to replace slim.batch_norm and successfully adapt it into the official DeepLab implementation, https://github.com/tensorflow/models/tree/master/research/deeplab.
It works fine with tensorflow<=1.12, However, recently I found when I tried tf=1.13, the computed loss would become Nan in a few steps after the start of the training.
When I switch sync_bn.batch_norm back to slim.batch_norm, every thing would be fine, although I no longer have the benefit from sync bn.

I tried many possibilities but still couldn't fix it and I wonder if you have some insights on it. I upload my version of codes and most of my implementation lie in this particular file:
https://github.com/tabrisweapon/deeplab3plus_tf_syncbn/blob/master/research/deeplab/sync_bn.py

Thx~

**Comment by ppwwyyxx** [comment]
This isn't related to tensorpack, so closing. I've run the object detection example with SyncBN and TF 1.13 without any issues.
